x-pack/plugins/enterprise_search/public/plugin.ts

```diff
@@ -153,7 +156,9 @@ export class EnterpriseSearchPlugin implements Plugin {
     }
   }
 
-  public start() {}
+  public start(core: CoreStart) {
+    docLinksService.setDocLinks(core.docLinks);
+  }
```

Comment:
```suggestion
  public start(core: CoreStart) {
    // This must be called here in start() and not in `applications/index.tsx` to prevent loading
    // race conditions with our apps' `routes.ts` being initialized before `renderApp()`
    docLinksService.setDocLinks(core.docLinks);
  }
```

In light of our previous discussion, do you mind adding a quick comment here noting that this cannot be done in index.tsx but must be done before our apps are loaded in order to prevent race conditions?